OptimizelySDK/Odp/RestApiManager.cs

```diff
@@ -0,0 +1,164 @@
+        /// <param name="events">ODP events to send</param>
+        /// <returns>Retry is true - if network or server error (5xx), otherwise false</returns>
+        public bool SendEvents(string apiKey, string apiHost, List<OdpEvent> events)
+        {
+            if (string.IsNullOrWhiteSpace(apiKey) || string.IsNullOrWhiteSpace(apiHost))
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{EVENT_SENDING_FAILURE_MESSAGE} (Parameters apiKey or apiHost invalid)");
+                return false;
+            }
+
+            if (events.Count == 0)
+            {
+                _logger.Log(LogLevel.ERROR, $"{EVENT_SENDING_FAILURE_MESSAGE} (no events)");
+                return false;
+            }
+
+            var endpoint = $"{apiHost}/v3/events";
+            var data = JsonConvert.SerializeObject(events);
+            var shouldRetry = false;
+
+            HttpResponseMessage response = default;
+            try
+            {
+                response = SendEventsAsync(apiKey, endpoint, data).GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                _errorHandler.HandleError(ex);
+                _logger.Log(LogLevel.ERROR, $"{EVENT_SENDING_FAILURE_MESSAGE} (network error)");
+                shouldRetry = true;
+            }
+
+            var responseStatusCode = response == null ? 0 : (int)response.StatusCode;
+            if (responseStatusCode >= 400)
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{EVENT_SENDING_FAILURE_MESSAGE} (${responseStatusCode})");
+            }
+
+            if (responseStatusCode >= 500)
+            {
+                shouldRetry = true;
+            }
+
+            return shouldRetry;
+        }
+
+        /// <summary>
+        /// Async call to send the event data to ODP
+        /// </summary>
+        /// <param name="apiKey">ODP public key</param>
+        /// <param name="endpoint">Fully-qualified ODP REST API endpoint</param>
+        /// <param name="data">JSON string version of ODP event data</param>
+        /// <returns>HTTP response endpoint</returns>
+        private async Task<HttpResponseMessage> SendEventsAsync(string apiKey, string endpoint,
+            string data
+        )
+        {
+            var request = BuildRequestMessage(apiKey, endpoint, data);
+
+            var response = await _httpClient.SendAsync(request);
+
+            return response;
+        }
+
+        /// <summary>
+        /// Build an HTTP request message
+        /// </summary>
+        /// <param name="apiKey">ODP public key</param>
+        /// <param name="endpoint">Fully-qualified ODP REST API endpoint</param>
+        /// <param name="data">JSON string version of ODP event data</param>
+        /// <returns>Ready request message to send</returns>
+        private static HttpRequestMessage BuildRequestMessage(string apiKey, string endpoint,
```

Comment:
What's the point to add this method. GraphQL is almost same. Not sure, please explain, couldn't catch if any diff in this method.

Reply:
There's a bit of a difference in the payload for sending an ODP event to the REST APi and a query to the GraphQL endpoint. 

Happy to walk through the diffs if ya want.  

It was a close similarity.  I'm renaming the two methods for clarity.